From f91c824355e966034148aeb47ca0e265d10de99e Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 18 Feb 2024 12:33:21 +0200
Subject: [PATCH] [FSW-2072] zephyr: include fs1 boot info hdr

Include dev specific boot_info header for FS1, which is located under
the fs/fs1 folder.

This patch also removes the duplication of including this header
separately for every component, and puts the include in 1 place.

Change-Id: I9060e955bb98aa0a84654460edf9ebb7267bc2fd
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 25bb3cc..ec0a27a 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -29,6 +29,12 @@
 	ZEPHYR_CONF_DEV_EXTRA_CFLAGS += -DTARGET_IS_ONIX
 endif
 
+ifeq ($(DEV_PREFIX),fs1)
+ZEPHYR_CONF_DEV_EXTRA_CFLAGS += -include $(EMBEDDED_ROOT)/specs/fs/$(DEVICE_NAME)/$(DEVICE_NAME)_boot_info.h
+else
+ZEPHYR_CONF_DEV_EXTRA_CFLAGS += -include $(EMBEDDED_ROOT)/specs/$(DEVICE_NAME)/$(DEVICE_NAME)_boot_info.h
+endif
+
 DEVICE_NAME = $(firstword $(subst _, ,$(HL_DEV)))
 
 # zephyr uses west tool to perform repo management, installation
@@ -109,7 +115,7 @@
 ZEPHYR_MGMT_DIR = build-mgmt
 ZEPHYR_MGMT_NAME = zephyr_arc_mgmt
 ZEPHYR_CONF_DEFAULT ?= $(ZEPHYR_MGMT_DIR)
-ZEPHYR_CONF_OPTS += -DEXTRA_CFLAGS="-Werror -Wno-misleading-indentation -include $(EMBEDDED_ROOT)/specs/$(DEVICE_NAME)/$(DEVICE_NAME)_boot_info.h $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
+ZEPHYR_CONF_OPTS += -DEXTRA_CFLAGS="-Werror -Wno-misleading-indentation $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
 ZEPHYR_CONF_OPTS_MGMT += -DBOARD=hl_$(DEVICE_NAME)_mgmt_$(HL_TARGET)
 ZEPHYR_CONF_OPTS_MGMT += -DDEVICE=$(DEVICE_NAME)
 ZEPHYR_MGMT_SRCDIR = $(EMBEDDED_DIR)/zephyr-apps/arcmgmt
@@ -163,7 +169,7 @@
 endif #PREBOOT_RUN_FROM_ICCM
 #TODO - TARGET_IS_ASIC: Need to check why on ZEPHYR_CONF_OPTS its not working as expected
 ifeq ($(HL_TARGET),asic)
-ZEPHYR_CONF_OPTS_PREBOOT += -DEXTRA_CFLAGS="-include $(EMBEDDED_ROOT)/specs/$(DEVICE_NAME)/$(DEVICE_NAME)_boot_info.h $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
+ZEPHYR_CONF_OPTS_PREBOOT += -DEXTRA_CFLAGS="$(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
 endif
 ZEPHYR_CONFIGURE_CMDS += && $(ZEPHYR_CONF_ENV) $(HOST_DIR)/bin/cmake $(ZEPHYR_CONF_OPTS) $(call CREATE_APP_VERSION,preboot,$(ZEPHYR_PREBOOT_DIR)) $(ZEPHYR_CONF_OPTS_PREBOOT) -B $(ZEPHYR_PREBOOT_DIR)
 ZEPHYR_BUILD_CMDS += && $(TARGET_MAKE_ENV) $(MAKE) -C $(@D)/$(ZEPHYR_PREBOOT_DIR)
@@ -210,11 +216,11 @@
 ZEPHYR_FW_LOADER_NAME = zephyr_agent_fw_loader
 ZEPHYR_CONF_DEFAULT ?= $(ZEPHYR_FW_LOADER_DIR)
 ifdef BR2_arcle
-ZEPHYR_CONF_OPTS += -DEXTRA_CFLAGS="-Werror -Wno-misleading-indentation -include $(EMBEDDED_ROOT)/specs/$(DEVICE_NAME)/$(DEVICE_NAME)_boot_info.h $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
+ZEPHYR_CONF_OPTS += -DEXTRA_CFLAGS="-Werror -Wno-misleading-indentation $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
 ZEPHYR_CONF_OPTS_AGENT_FW_LOADER += -DBOARD=hl_$(DEVICE_NAME)_agent_$(HL_TARGET)
 endif
 ifdef BR2_aarch64
-ZEPHYR_CONF_OPTS += -DEXTRA_CFLAGS="-mstrict-align -Werror -Wno-misleading-indentation -include $(EMBEDDED_ROOT)/specs/$(DEVICE_NAME)/$(DEVICE_NAME)_boot_info.h $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
+ZEPHYR_CONF_OPTS += -DEXTRA_CFLAGS="-mstrict-align -Werror -Wno-misleading-indentation $(ZEPHYR_CONF_DEV_EXTRA_CFLAGS)"
 ZEPHYR_CONF_OPTS_AGENT_FW_LOADER += -DBOARD=hl_$(DEVICE_NAME)_$(HL_TARGET)
 endif
 ZEPHYR_CONF_OPTS_AGENT_FW_LOADER += -DDEVICE=$(DEVICE_NAME)
